src/synth/synth-context.adb

```diff
@@ -358,9 +358,11 @@ package body Synth.Context is
                   Value2net (Val, 1, V, Res);
                   return Res;
                end;
-            elsif Val.Typ.Drange.W <= 32 then
-               return Build_Const_UB32
-                 (Build_Context, Uns32 (Val.Scal), Val.Typ.Drange.W);
+            elsif Val.Scal <= Int64(Uns32'Last) then
+               return Build_Const_UB32 (
+                 Build_Context,
+                 Uns32 (Val.Scal),
+                 Width'min(32, Val.Typ.Drange.W));
             else
                raise Internal_Error;
             end if;
```

Comment:
Can you create a different PR for this ?  I fear there is an issue somewhere else; I should investigate.